Mathlib/RingTheory/TwoSidedIdeal/Operations.lean

```diff
@@ -88,16 +88,40 @@ variable [NonUnitalRingHomClass F R S]
 
 /--
 Preimage of a two-sided ideal, as a two-sided ideal. -/
-def comap (I : TwoSidedIdeal S) : TwoSidedIdeal R where
-  ringCon := I.ringCon.comap f
+def comap : TwoSidedIdeal S →o TwoSidedIdeal R where
+  toFun I := ⟨I.ringCon.comap f⟩
+  monotone' := by
+    intro I J h
+    rw [le_iff] at h
+    intro x
+    specialize @h (f x)
+    simpa [mem_iff, RingCon.comap]
 
 lemma mem_comap {I : TwoSidedIdeal S} {x : R} :
     x ∈ I.comap f ↔ f x ∈ I := by
   simp [comap, RingCon.comap, mem_iff]
 
+/--
+If `R` and `S` are isomorphic as rings, then two-sided ideals of `R` and two-sided ideals of `S` are
+order isomorphic.
+-/
+def orderIsoOfRingEquiv (e : R ≃+* S) : TwoSidedIdeal R ≃o TwoSidedIdeal S :=
+  OrderIso.ofHomInv (comap e.symm) (comap e) (by ext; simp [mem_comap])
+    (by ext; simp [mem_comap])
 
 end NonUnitalNonAssocRing
 
+section NonAssocRing
+
+variable {R S T : Type*}
+variable [NonAssocRing R] [NonAssocRing S] [NonAssocRing T]
+
+lemma comap_comp (I : TwoSidedIdeal T) (f : R →+* S) (g : S →+* T) :
+    I.comap (g.comp f) = (I.comap g).comap f := by
```

Comment:
My first thought was that the LHS is simpler here (and so maybe it would be better if it was the RHS), but I'm not sure.

Reply:
I will listen to you and change the name to `comap_comap`